File: frontend/src/models/encounter/ParticipanceDataDto.ts
import { Field } from "../map/Board";
import { CharacterItem } from "../character";

export type ParticipanceDataDto = {
  id: number;
  initiativeOrder: number;
  isSurprised: boolean;
  numberOfActionsTaken: number;
  numberOfBonusActionsTaken: number;
  numberOfAttacksTaken: number;
  distanceTraveled: number;
  character: CharacterItem;
  occupiedField: Field;
};

File: pracadyplomowa/Models/DTOs/Encounter/ParticipanceDataDto.cs
using pracadyplomowa.Models.DTOs.Map.Field;
using pracadyplomowa.Models.Entities.Characters;

namespace pracadyplomowa.Models.DTOs.Encounter;

public record ParticipanceDataDto
{
    public int InitiativeOrder { get; set; }
    public bool IsSurprised { get; set; }
    public int NumberOfActionsTaken { get; set; }
    public int NumberOfBonusActionsTaken { get; set; }
    public int NumberOfAttacksTaken { get; set; }
    public int DistanceTraveled { get; set; }
    public ParticipanceCharacterSummaryDto Character { get; set; }
    public ICollection<FieldDto> OccupiedFields { get; set; }
}